iree/turbine/kernel/wave/unrolling.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from .utils.symbol_utils import subs_idxc
+from .constraints import Constraint
+from .utils.general_utils import get_tiling_constraint
+from iree.turbine.kernel._support.tracing import CapturedTrace
+import torch.fx as fx
+from ..ops.wave_ops import Iterate, Output, Placeholder, get_custom
+
+
+def remap_iter_args(
+    iter_args: list[fx.Node], output: Output, value_use_map: dict[fx.Node, fx.Node]
+) -> None:
+    """
+    Add a mapping of iter_args to the output return values in order to keep track
+    of how uses need to be updated for unrolled iterations.
+    """
+    for iter_arg, return_val in zip(iter_args, output.return_vals[0]):
+        value_use_map[iter_arg] = return_val
+
+
+def unroll(
+    iterate: Iterate,
+    unroll_factor: int,
+    trace: CapturedTrace,
+    constraints: list[Constraint],
+) -> None:
+    """
+    Unroll an iterate node in the graph `unroll_factor` times.
+    This is done by creating `unroll_factor` copies of the iteration body
+    and adjusting the step size and boundaries accordingly.
```

Comment:
Could you add a comment describing the value mapping and the need to use the original nodes as opposed to all nodes?